hs_restclient/__init__.py

```diff
@@ -1 +1,666 @@
+        :param from_date: Filter results to those created after from_date.  Must be datetime.date.
+        :param to_date: Filter results to those created before to_date.  Must be datetime.date.  Because dates have
+            no time information, you must specify date+1 day to get results for date (e.g. use 2015-05-06 to get
+            resources created up to and including 2015-05-05)
+        :param types: Filter results to particular HydroShare resource types.  Must be a sequence type
+            (e.g. list, tuple, etc.), but not a string.
+
+        :raises: HydroShareHTTPException to signal an HTTP error
+        :raises: HydroShareArgumentException for any invalid arguments
+
+        :return: A generator that can be used to fetch dict objects, each dict representing
+            the JSON object representation of the resource returned by the REST end point.  For example:
+
+        >>> for resource in hs.getResourceList():
+        >>>>    print resource
+         {u'bag_url': u'http://www.hydroshare.org/static/media/bags/e62a438bec384087b6c00ddcd1b6475a.zip',
+          u'creator': u'B Miles',
+          u'date_created': u'05-05-2015',
+          u'date_last_updated': u'05-05-2015',
+          u'resource_id': u'e62a438bec384087b6c00ddcd1b6475a',
+          u'resource_title': u'My sample DEM',
+          u'resource_type': u'RasterResource',
+          u'science_metadata_url': u'http://www.hydroshare.org/hsapi/scimeta/e62a438bec384087b6c00ddcd1b6475a/',
+          u'public': True}
+         {u'bag_url': u'http://www.hydroshare.org/static/media/bags/hr3hy35y5ht4y54hhthrtg43w.zip',
+          u'creator': u'B Miles',
+          u'date_created': u'01-02-2015',
+          u'date_last_updated': u'05-13-2015',
+          u'resource_id': u'hr3hy35y5ht4y54hhthrtg43w',
+          u'resource_title': u'Other raster',
+          u'resource_type': u'RasterResource',
+          u'science_metadata_url': u'http://www.hydroshare.org/hsapi/scimeta/hr3hy35y5ht4y54hhthrtg43w/',
+          u'public': True}
+
+
+          Filtering (have):
+
+          /hsapi/resourceList/?from_date=2015-05-03&to_date=2015-05-06
+          /hsapi/resourceList/?user=admin
+          /hsapi/resourceList/?owner=admin
+          /hsapi/resourceList/?creator=admin
+          /hsapi/resourceList/?group=groupname
+          /hsapi/resourceList/?types=GenericResource&types=RasterResource
+
+          Filtering (need):
+
+          /hsapi/resourceList/?sharedWith=user
+
+        :raises:: HydroShareArgumentException if any filter parameters are invalid.
+        """
+        url = "{url_base}/resourceList/".format(url_base=self.url_base)
+
+        params = {}
+        if creator:
+            params['creator'] = creator
+        if owner:
+            params['owner'] = owner
+        if user:
+            params['user'] = user
+        if group:
+            params['group'] = group
+        if from_date:
+            if type(from_date) is datetime.date:
```

Comment:
Should be using `isinstance()` for type checking, if you really need type checking.

But also, maybe don't type check. It'll already fail looking for `strftime()` so the extra code to type check is largely redundant, and also makes testing harder (maybe some needs to mock out this parameter to test something).